src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptions.cs

```diff
@@ -181,5 +194,51 @@ public Action<RoslynCompilationContext> CompilationCallback
                 _compilationCallback = value;
             }
         }
+
+        /// <summary>
+        /// Gets or sets a value that determines if Razor files (Razor Views and Razor Pages) are recompiled and updated 
+        /// if files change on disk.
+        /// <para>
+        /// When <see langword="true"/>, MVC will use <see cref="IFileProvider.Watch(string)"/> to watch for changes to 
+        /// Razor files in configured <see cref="IFileProvider"/> instances.
+        /// </para>
+        /// </summary>
+        /// <value>
+        /// The default value is <see langword="true"/> if the version is
+        /// <see cref="CompatibilityVersion.Version_2_2"/> or later and <see cref="IHostingEnvironment.EnvironmentName"/> is <c>Development</c>; 
+        /// <see langword="false"/> otherwise.
```

Comment:
Not complicated enough 😈:
```
The default value is <see langword="true"/> if the version is
<see cref="CompatibilityVersion.Version_2_1"/> or earlier. If the version is later and
<see cref="IHostingEnvironment.EnvironmentName"/> is <c>Development</c>,
the default value is also <see langword="true"/>. Otherwise, the default value is
<see langword="false"/>.
```